pkg/lua/contrib/lua_run.c

```diff
@@ -0,0 +1,275 @@
+  {LUA_UTF8LIBNAME, luaopen_utf8},
+  {LUA_DBLIBNAME, luaopen_debug},
+};
+
+LUALIB_API int luaR_openlibs(lua_State *L, uint16_t modmask)
+{
+    int lib_index;
+
+    #ifdef LUA_DEBUG
+        luaL_requiref(L, LUA_TESTLIBNAME, luaB_opentests, 1);
+        lua_pop(L, 1);
+    #endif
+
+    for (lib_index = 0; lib_index < LUAR_LOAD_O_ALL;
+         lib_index++, modmask >>= 1) {
+        const luaL_Reg *lib = loadedlibs + lib_index;
+
+        if (!(modmask & 1)) {
+            continue;
+        }
+        /* TODO: how can the loading fail? */
+        luaL_requiref(L, lib->name, lib->func, 1);
+        lua_pop(L, 1);  /* remove lib from stack (it is already global) */
+    }
+
+    return lib_index;
+}
+
+/**
+ * Jump back to a save point (defined with setjmp).
+ *
+ * @note    This function never returns!
+ */
+NORETURN static int _jump_back (lua_State *L)
+{
+    jmp_buf *jump_buffer = *(jmp_buf **)lua_getextraspace(L);
+
+    /* FIXME: I dont think it's OK to print a message */
+    lua_writestringerror("PANIC: unprotected error in call to Lua API (%s)\n",
+                        lua_tostring(L, -1));
+
+    /* TODO: try to return some info about the error object. */
+
+    longjmp(*jump_buffer, 1);
+}
+
+static int luaR_do_module_or_buf(const char *buf, size_t buflen,
+                            const char *modname, void *memory, size_t mem_size,
+                            uint16_t modmask, int *retval)
+{
+    jmp_buf jump_buffer;
+    lua_State * volatile L = NULL;
+    volatile int tmp_retval = 0; /* we need to make it volatile because of the
+                                    setjmp/longjmp */
+    volatile int status = LUAR_EXIT;
+    int compilation_result;
+
+    if(setjmp(jump_buffer)) { /* We'll teleport back here if something goes wrong*/
+        status = LUAR_INTERNAL_ERR;
+        goto luaR_do_error;
+    }
+
+    L = luaR_newstate(memory, mem_size, _jump_back);
+    if (L == NULL) {
+        status = LUAR_STARTUP_ERR;
+        goto luaR_do_error;
+    }
+
+    /* lua_getextraspace() gives us a pointer to an are large enough to hold a
+     * pointer.
+     *
+     * We store a pointer to the jump buffer in that area.
+     *
+     * lua_getextraspace() is therefore a pointer to a pointer to jump_buffer.
+     */
+    *(jmp_buf **)lua_getextraspace(L) = &jump_buffer;
+
+    tmp_retval = luaR_openlibs(L, modmask);
+    if (tmp_retval != LUAR_LOAD_O_ALL) {
+        status = LUAR_LOAD_ERR;
+        goto luaR_do_error;
+    }
+
+    if (buf == NULL) {
+        compilation_result = luaR_getloader(L, modname);
+    } else {
+        compilation_result = luaL_loadbufferx(L, buf, buflen, modname, "t");
+    }
+
+    switch (compilation_result) {
+    case LUAR_MODULE_NOTFOUND:
+        status = LUAR_NOMODULE;
+        goto luaR_do_error;
+    case LUA_ERRSYNTAX:
+        status = LUAR_COMPILE_ERR;
+        goto luaR_do_error;
+    case LUA_ERRMEM: /* fallthrough */
+    case LUA_ERRGCMM:
+    default:
```

Comment:
please also uncrustify. here and throughout the rest of the submitted code.

Reply:
Ok. One thing I'll keep, though, is the indentation of the code between `#ifdef`s. I find it quite hard to follow conditionals if they are not indented, and I don't see how a compile-time conditional would make a difference.